application-example.conf

```diff
@@ -1,133 +1,192 @@
+
+  # elastic search
+  elastic {
+    url: "http://localhost:9200"
+    index: "lucille-default"
+    type: "lucille-type"
+    sendEnabled: false
+  }
+
+  # solr basic indexer config
+  solr {
+    url: "http://localhost:8983/solr/callbacktest"
+  }
+
+  # solr cloud mode with url
+  solr {
+    useCloudClient: true
+    defaultCollection: "zktest2"
+    url: ["http://localhost:8983/solr"]
+  }
+
+  # solr cloud mode with zkHosts
+  solr {
+    useCloudClient: true
+    defaultCollection: "gettingstarted"
+    zkHosts: ["zookeeper1:2181", "zookeeper2:2181", "zookeeper3:2181"]
+    # optional
+    zkChroot: "/solr"
+  }
+
+  # zookeeper settings for solr indexer
+  zookeeper {
+    connectString: "localhost:2181"
+  }
+
+  # open search
+  opensearch {
+    url: "https://admin:admin@localhost:9200"
+    url: ${?OPENSEARCH_URL} # allow env override
+    index: "lucille-default"
+    index: ${?OPENSEARCH_INDEX}
+    acceptInvalidCert: false # only enable for testing ssl/https against localhost
+  }
+## ----------------------------------------------------------------------------------------------------------------------------------
+
+
+## Worker Config: -----------------------------------------------------------------------------------------------------------------
+  worker {
+    # maximum number of times across all workers that an attempt should be made to process any given document;
+    # when this property is not provided, retries will not be tracked and no limit will be imposed;
+    # this property requires that zookeeper is available at the specified zookeeper.connectString
+    maxRetries: 2
+
+    # number of worker threads to start for each pipeline when running lucille in local / single-JVM mode
+    threads: 2
+
+    # maximum time to spend processing a message, after which the worker will assume a problem and shut down
+    # NOTE: not supported while com.kmwllc.lucille.core.Worker and PipelineWorker are being reconciled
+    maxProcessingSecs: 600 # 10 minutes
+
+    # name of pipeline
+    pipeline: "pipeline_name"
+  }
+## ----------------------------------------------------------------------------------------------------------------------------------
+
+
+## Kafka Config: -----------------------------------------------------------------------------------------------------------------
   kafka {
     bootstrapServers: "localhost:9092"
+
+    # how long a kafka poll should wait for data before returning an empty record set
+    pollIntervalMs: 250
+
+    # maximum time allowed between kafka polls before consumer is evicted from consumer group
+    maxPollIntervalSecs: 600 # 10 minutes
+
+    # ID of consumer group that all lucille workers should belong to
+    consumerGroupId: "lucille_workers"
+
+    maxRequestSize: 250000000
+
+    securityProtocol: "SSL"
+
+    consumerPropertyFile: ".../consumer-conf/consumer.properties"
+
+    producerPropertyFile: ".../consumer-conf/producer.properties"
+
+    documentDeserializer: # TODO
```

Comment:
Am unsure what these items are doing, I see that KafkaUtils.java is placing them in a properties map that is then fed into the AdminUser creation, but not sure what a good example for these properties might be.

Reply:
Follow up: Might need to add tests to cover a lot of these properties, as they currently don't exist

Reply:
Resolved